File: example/handwritten-src/objc/TXSTextboxListenerImpl.m
#import "TXSTextboxListenerImpl.h"
#import "TXSItemList.h"

@implementation TXSTextboxListenerImpl {
    UITextView * _textView;
}

- (id)initWithUITextView:(UITextView *)textView
{
    if (self = [super init]) {
        _textView = textView;
    }
    return self;
}

- (void)update:(TXSItemList *)items
{
    NSString *str = [items.items componentsJoinedByString:@"\n"];
    _textView.text = str;
}

@end

File: example/handwritten-src/swift/TXSTextboxListenerImpl.swift
//
//  TXSTextboxListenerImpl.swift
//  TextSort
//
//  Created by Bruno Coelho on 10/03/2017.
//  Copyright © 2017 Dropbox, Inc. All rights reserved.
//

import UIKit

@objc class TXSTextboxListenerImpl : NSObject, TXSTextboxListener {
    private var textView_: UITextView
    
    @objc(initWithUITextView:)
    init (textView: UITextView) {
        textView_ = textView
    }
    
    @objc func update(_ items: TXSItemList) {
        textView_.text = items.items.joined(separator: "\n")
    }
}
